### DISCUSSION TITLE
Is a custom androidScheme deprecated in Capacitor 6?

### DISCUSSION THREAD

**Discussion by neave** [ionic-team/capacitor#7431] [Q&A] [answered]
We're using a custom `androidScheme` in our Capacitor project. Since upgrading to Capacitor 6 we are unable to compile, as Android Studio throws an error regarding `allowedOriginRules`.

Is it no longer possible to set a custom `androidScheme` in Capacitor 6? This is still possible on iOS with the `iosScheme`. Thanks.

**Comment by jcesarmobile** [comment, marked as answer]
It's a bug, in Capacitor 6 there is a new javascript injector that requires to provide the url where the javascript execution is allowed, and it's allowed for the app url, i.e. `https://localhost`, but for google, if using a custom scheme, the hostname should not be included in the url, only the scheme according to this:
https://developer.android.com/reference/androidx/webkit/WebViewCompat#addWebMessageListener(android.webkit.WebView,java.lang.String,java.util.Set%3Cjava.lang.String%3E,androidx.webkit.WebViewCompat.WebMessageListener)

So yeah, something we will need to fix, would be good if you could create an issue and provide a sample app that reproduces the crash.

But anyway, you should know that Google engineers discourage using custom schemes on android, they recommend only using https.

Custom schemes also cause problems starting in Chrome 117 if using History API 
https://ionic.io/blog/capacitor-android-customscheme-issue-with-chrome-117
